internal/pkg/vm/internal/vmcontext/vmcontext.go

```diff
@@ -368,6 +389,10 @@ func (vm *VM) applyMessage(msg *types.UnsignedMessage, onChainMsgSize uint32, rn
 	// - sender balance withheld
 	// - (if did not previously exist, the new auto-created target actor)
 
+	if err := vm.state.SetActor(vm.context, msg.From, fromActor); err != nil {
+		panic(err)
+	}
+
```

Comment:
This behaviour was reverted from the spec, so the comment above (copied from a prior version) is suggesting the wrong thing. The spec interpreter now says:

```
// At this point, construct compTreePreSend as a state snapshot which includes
// the sender paying gas, and the sender's CallSeqNum being incremented;
// at least that much state change will be persisted even if the
// method invocation subsequently fails.
```

Reply:
cc @frrist for messages tests, can you check that the implicit destination actor is not created in this case?